impress-org/givewp
# Feature: save title as pageslug on publish

Resolves Asana task: https://app.asana.com/0/1205428662243164/1205735215345414/f

## Description

This PR saves the form title as the page slug when a form first publishes. After a form has been published the page slug can still be edited, however it is not respective of the form title on any subsequent change.

If the form title is not updated on the first publish the page slug will continue to to work as it previously did, generating the slug trailing with a new number.

I had looked into a central way of dispatching this in the onSave function call - however had inconsistent state changes and the page slug was being overwritten on refresh.  Due to this I've opted to include conditional dispatch call's for the page slug on both form title controls.

Re: https://lw.slack.com/archives/C05QFUQ805V/p1697494770284989?thread_ts=1694639392.259359&cid=C05QFUQ805V

## Visuals


https://github.com/impress-org/givewp/assets/75056371/e5eb541a-b730-4ac5-bbc5-4bd127d91163



## Testing Instructions

- Create a form
- Update the form title.
- Publish the form.
- View new page slug.
- Visit page - verify pageslug works.

## Pre-review Checklist




---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205799200925606
